fetcher/fetcher.go

```diff
@@ -94,6 +98,28 @@ func (bot *fetcherImpl) setError(err error) {
 	bot.errmu.Unlock()
 }
 
+func (bot *fetcherImpl) processQueue() {
+	for {
+		select {
+		case <-bot.ctx.Done():
```

Comment:
Should you `close(bot.blockQueue)` here?

Reply:
It's a good point. I changed the code so that `processQueue()` reads from the channel until it's closed, and we close it when `Run()` exits.